### PR TITLE
x-pack/filebeat/input/http_endpoint: allow endpoint to receive PUT and PATCH requests

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

See title.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ] The approach taken allows specification of the method that is accepted. A more relaxed approach would allow any of PUT/PATCH/POST.

## Related issues

- For elastic/integrations#8066

### CODE REVIEW

File: x-pack/filebeat/input/http_endpoint/input_test.go [hunk 1]
```diff
@@ -55,6 +56,60 @@ var serverPoolTests = []struct {
...
+				ListenPort:    "9001",
+				URL:           "/",
+				Prefix:        "json",
+				ContentType:   "application/json",
+			},
+		}},
+		events: []target{
+			{url: "http://127.0.0.1:9001/", event: `{"a":1}`},
+			{url: "http://127.0.0.1:9001/", event: `{"b":2}`},
+			{url: "http://127.0.0.1:9001/", event: `{"c":3}`},
```
Comment (bhapas) [inline x-pack/filebeat/input/http_endpoint/input_test.go:78]: How does it behave if there is no request body for a `PUT`

File: x-pack/filebeat/input/http_endpoint/input_test.go [hunk 2]
```diff
@@ -288,6 +343,18 @@ func TestServerPool(t *testing.T) {
...
 
+func doRequest(method, url, contentType string, body io.Reader) (*http.Response, error) {
+	if method == "" {
+		method = http.MethodPost
+	}
+	req, err := http.NewRequest(method, url, body)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", contentType)
```
Comment (bhapas) [inline x-pack/filebeat/input/http_endpoint/input_test.go:354]: Do we need to check `Content-Length > 0` to make sure we always `request.body`?

File: x-pack/filebeat/docs/inputs/input-http-endpoint.asciidoc [hunk 3]
```diff
@@ -279,6 +279,12 @@ This option defines the provider of the webhook that uses CRC (Challenge-Respons
 
 The secret token provided by the webhook owner for the CRC validation. It is required when a `crc.provider` is set.
 
+[float]
+==== `method`
+
+The HTTP method handled by the endpoint. If specified, `method` must be `POST`, `PUT` or `PATCH`.
+The default method is `POST`. If `PUT` or `PATCH` are specified, requests of using those method types are accepted, but are treated as `POST` requests and are expected to have a request body containing the request data.
```
[the remark this answers is not in the document]
Comment (efd6) [reply x-pack/filebeat/docs/inputs/input-http-endpoint.asciidoc:286]: The size of {`PUT`, `PATCH`} is not unity, so plural is correct here.

File: x-pack/filebeat/input/http_endpoint/input_test.go (hunk 1 above)
Comment (efd6) [reply x-pack/filebeat/input/http_endpoint/input_test.go:78]: Exactly the same way that a zero body POST would behave.

File: x-pack/filebeat/input/http_endpoint/input_test.go (hunk 2 above)
Comment (efd6) [reply x-pack/filebeat/input/http_endpoint/input_test.go:354]: This doesn't change the behaviour at all except for allowing PUT and PATCH requests to be used, so I don't think so.

**Review by andrewkroh — APPROVED** [review]
LGTM.

**Pull request merged**